src/control/positionControl/PositionControl.cpp

```diff
@@ -30,10 +27,12 @@ RobotCommand PositionControl::computeAndTrackPath(const world::Field &field, int
     return command;
 }
 
-bool PositionControl::shouldRecalculatePath(const Vector2 &currentPosition, const Vector2 &targetPos, int robotId) {
+bool PositionControl::shouldRecalculatePath(const Vector2 &currentPosition, const Vector2 &targetPos,
+                                            const Vector2 &currentVelocity, int robotId) {
     return computedPaths[robotId].empty() || PositionControlUtils::isTargetChanged(targetPos, computedPaths[robotId].back()) ||
-           collisionDetector.isRobotCollisionBetweenPoints(currentPosition, computedPaths[robotId].front());
+            (currentVelocity != Vector2() && collisionDetector.getRobotCollisionBetweenPoints(currentPosition, computedPaths[robotId].front()));
```

Comment:
That's the thing, because of machine errors the length is never zero. However, when comparing to a different vector, like (0, 0), it takes the error into account. I will modify to Vector2(0, 0) for clarity